pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -156,19 +160,21 @@ func (r *Reconciler) deleteUponCompletion(ctx context.Context, o results.Object)
 		return controller.NewRequeueAfter(requeueAfter)
 	}
 
-	logger.Infow("Deleting object", zap.String("results.tekton.dev/uid", string(o.GetUID())))
+	logger.Infow("Deleting object", zap.String("results.tekton.dev/uid", string(o.GetUID())),
+		zap.Int64("results.tekton.dev/time-taken-seconds", int64(time.Since(*completionTime).Seconds())))
 	if err := r.objectClient.Delete(ctx, o.GetName(), metav1.DeleteOptions{
 		Preconditions: metav1.NewUIDPreconditions(string(o.GetUID())),
 	}); err != nil && !errors.IsNotFound(err) {
-		logger.Errorw("Error deleting object", zap.Error(err))
-		return err
+		logger.Debugw("Error deleting object", zap.Error(err))
+		return fmt.Errorf("error deleting object: %w", err)
 	}
 
+	logger.Debugw("Object has been successfully deleted", zap.Int64("results.tekton.dev/time-taken-seconds", int64(time.Since(*completionTime).Seconds())))
 	return nil
 }
 
 func isDone(o results.Object) bool {
-	return o.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue()
+	return !o.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsUnknown()
```

Comment:
Yikes...could we add a test for this so this sort of thing does not go undetected?